File: src/main/java/io/myosotisdev/utopianism/serialize/ParticleTagSerializer.java
package io.myosotisdev.utopianism.serialize;

import io.myosotisdev.utopianism.modules.effect.ParticleEffect;
import net.minestom.server.particle.Particle;
import org.jglrxavpok.hephaistos.nbt.NBTCompound;

import java.util.function.Function;

public class ParticleTagSerializer extends Serializer<ParticleEffect>
{
    @Override
    public String serialize(ParticleEffect obj)
    {
        return null;
    }

    @Override
    public ParticleEffect deserialize(String string)
    {
        String[] args = string.split(",");
        return null;
    }
}

File: src/main/java/io/myosotisdev/utopianism/serialize/ParticleTagSerializer.kt
package io.myosotisdev.utopianism.serialize

import io.myosotisdev.utopianism.modules.effect.ParticleEffect

class ParticleTagSerializer : Serializer<ParticleEffect?>()
{
    override fun serialize(obj: ParticleEffect?): String?
    {
        return null
    }

    override fun deserialize(string: String?): ParticleEffect?
    {
        val args = string!!.split(",".toRegex())
                .dropLastWhile { it.isEmpty() }
                .toTypedArray()
        return ParticleEffect()
    }
}